arrow[core,mimalloc]:x64-windows Build failure

Package: arrow[core,mimalloc]:x64-windows -> 14.0.1#1

**Host Environment**

- Host: x64-windows
- Compiler: MSVC 19.37.32825.0
-    vcpkg-tool version: 2999-12-31-unknownhash
    vcpkg-scripts version: ecf95e1e7 2023-11-18 (4 days ago)

**To Reproduce**

`vcpkg x-test-features --ci-feature-baseline=scripts/ci.feature.baseline.txt --failure-logs=log --all`
**Failure logs**

```
-- Using cached apache-arrow-14.0.1.tar.gz.
-- Extracting source G:/v/downloads/apache-arrow-14.0.1.tar.gz
-- Applying patch msvc-static-name.patch
-- Applying patch utf8proc.patch
-- Applying patch thrift.patch
-- Applying patch fix-ci-error.patch
-- Using source at G:/v/b1/arrow/src/e-arrow-14-b060a9fe4a.clean
-- Found external ninja('1.11.0').
-- Configuring x64-windows
-- Building x64-windows-dbg
CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:134 (message):
    Command failed: G:/v/downloads/tools/cmake-3.27.1-windows/cmake-3.27.1-windows-i386/bin/cmake.exe --build . --config Debug --target install -- -v -j31
    Working Directory: G:/v/b1/arrow/x64-windows-dbg
    See logs for more information:
      G:\v\b1\arrow\install-x64-windows-dbg-out.log

Call Stack (most recent call first):
  vcpkg_installed/x64-windows/share/vcpkg-cmake/vcpkg_cmake_build.cmake:74 (vcpkg_execute_build_process)
  vcpkg_installed/x64-windows/share/vcpkg-cmake/vcpkg_cmake_install.cmake:16 (vcpkg_cmake_build)
  ports/arrow/portfile.cmake:66 (vcpkg_cmake_install)
  scripts/ports.cmake:170 (include)



```
<details><summary>G:\v\b1\arrow\install-x64-windows-dbg-out.log</summary>

```
Change Dir: 'G:/v/b1/arrow/x64-windows-dbg'

Run Build Command(s): "C:/Program Files/Microsoft Visual Studio/2022/Community/Common7/IDE/CommonExtensions/Microsoft/CMake/Ninja/ninja.exe" -v -v -j31 install
[1/193] cmd.exe /C "cd /D G:\v\b1\arrow\x64-windows-dbg && G:\v\downloads\tools\cmake-3.27.1-windows\cmake-3.27.1-windows-i386\bin\cmake.exe -Dcfgdir= -P G:/v/b1/arrow/x64-windows-dbg/mimalloc_ep-prefix/tmp/mimalloc_ep-mkdirs.cmake && G:\v\downloads\tools\cmake-3.27.1-windows\cmake-3.27.1-windows-i386\bin\cmake.exe -E touch G:/v/b1/arrow/x64-windows-dbg/mimalloc_ep-prefix/src/mimalloc_ep-stamp/mimalloc_ep-mkdir"
[2/193] cmd.exe /C "cd /D G:\v\b1\arrow\x64-windows-dbg\mimalloc_ep-prefix\src && G:\v\downloads\tools\cmake-3.27.1-windows\cmake-3.27.1-windows-i386\bin\cmake.exe -P G:/v/b1/arrow/x64-windows-dbg/mimalloc_ep-prefix/src/mimalloc_ep-stamp/mimalloc_ep-download-DEBUG.cmake && G:\v\downloads\tools\cmake-3.27.1-windows\cmake-3.27.1-windows-i386\bin\cmake.exe -E touch G:/v/b1/arrow/x64-windows-dbg/mimalloc_ep-prefix/src/mimalloc_ep-stamp/mimalloc_ep-download"
[3/193] cmd.exe /C "cd /D G:\v\b1\arrow\x64-windows-dbg && G:\v\downloads\tools\cmake-3.27.1-windows\cmake-3.27.1-windows-i386\bin\cmake.exe -E echo_append && G:\v\downloads\tools\cmake-3.27.1-windows\cmake-3.27.1-windows-i386\bin\cmake.exe -E touch G:/v/b1/arrow/x64-windows-dbg/mimalloc_ep-prefix/src/mimalloc_ep-stamp/mimalloc_ep-update"
[4/193] cmd.exe /C "cd /D G:\v\b1\arrow\x64-windows-dbg && G:\v\downloads\tools\cmake-3.27.1-windows\cmake-3.27.1-windows-i386\bin\cmake.exe -E echo_append && G:\v\downloads\tools\cmake-3.27.1-windows\cmake-3.27.1-windows-i386\bin\cmake.exe -E touch G:/v/b1/arrow/x64-windows-dbg/mimalloc_ep-prefix/src/mimalloc_ep-stamp/mimalloc_ep-patch"
[5/193] cmd.exe /C "cd /D G:\v\b1\arrow\x64-windows-dbg\mimalloc_ep-prefix\src\mimalloc_ep-build && G:\v\downloads\tools\cmake-3.27.1-windows\cmake-3.27.1-windows-i386\bin\cmake.exe -P G:/v/b1/arrow/x64-windows-dbg/mimalloc_ep-prefix/src/mimalloc_ep-stamp/mimalloc_ep-configure-DEBUG.cmake && G:\v\downloads\tools\cmake-3.27.1-windows\cmake-3.27.1-windows-i386\bin\cmake.exe -E touch G:/v/b1/arrow/x64-windows-dbg/mimalloc_ep-prefix/src/mimalloc_ep-stamp/mimalloc_ep-configure"
FAILED: mimalloc_ep-prefix/src/mimalloc_ep-stamp/mimalloc_ep-configure G:/v/b1/arrow/x64-windows-dbg/mimalloc_ep-prefix/src/mimalloc_ep-stamp/mimalloc_ep-configure 
cmd.exe /C "cd /D G:\v\b1\arrow\x64-windows-dbg\mimalloc_ep-prefix\src\mimalloc_ep-build && G:\v\downloads\tools\cmake-3.27.1-windows\cmake-3.27.1-windows-i386\bin\cmake.exe -P G:/v/b1/arrow/x64-windows-dbg/mimalloc_ep-prefix/src/mimalloc_ep-stamp/mimalloc_ep-configure-DEBUG.cmake && G:\v\downloads\tools\cmake-3.27.1-windows\cmake-3.27.1-windows-i386\bin\cmake.exe -E touch G:/v/b1/arrow/x64-windows-dbg/mimalloc_ep-prefix/src/mimalloc_ep-stamp/mimalloc_ep-configure"
CMake Error at G:/v/b1/arrow/x64-windows-dbg/mimalloc_ep-prefix/src/mimalloc_ep-stamp/mimalloc_ep-configure-DEBUG.cmake:4 (set):
  Syntax error in cmake code at

    G:/v/b1/arrow/x64-windows-dbg/mimalloc_ep-prefix/src/mimalloc_ep-stamp/mimalloc_ep-configure-DEBUG.cmake:4

  when parsing string

    G:/v/downloads/tools/cmake-3.27.1-windows/cmake-3.27.1-windows-i386/bin/cmake.exe;-DCMAKE_C_COMPILER=C:/Program Files/Microsoft Visual Studio/2022/Community/VC/Tools/MSVC/14.37.32822/bin/Hostx64/x64/cl.exe;-DCMAKE_CXX_COMPILER=C:/Program Files/Microsoft Visual Studio/2022/Community/VC/Tools/MSVC/14.37.32822/bin/Hostx64/x64/cl.exe;-DCMAKE_AR=C:/Program Files/Microsoft Visual Studio/2022/Community/VC/Tools/MSVC/14.37.32822/bin/Hostx64/x64/lib.exe;-DBUILD_SHARED_LIBS=OFF;-DBUILD_STATIC_LIBS=ON;-DBUILD_TESTING=OFF;-DCMAKE_BUILD_TYPE=DEBUG;-DCMAKE_CXX_FLAGS= /nologo /DWIN32 /D_WINDOWS  /utf-8 /GR /EHsc /MP  /D_SILENCE_TR1_NAMESPACE_DEPRECATION_WARNING;-DCMAKE_CXX_FLAGS_DEBUG=/D_DEBUG /MDd /Z7 /Ob0 /Od /RTC1  -MDd;-DCMAKE_CXX_FLAGS_MISIZEREL=/O1 /Ob1 /DNDEBUG -MD;-DCMAKE_CXX_FLAGS_RELEASE=/MD /O2 /Oi /Gy /DNDEBUG /Z7  -MD;-DCMAKE_CXX_FLAGS_RELWITHDEBINFO=/Z7 /O2 /Ob1 /DNDEBUG -MD;-DCMAKE_CXX_STANDARD=17;-DCMAKE_C_FLAGS= /nologo /DWIN32 /D_WINDOWS /W3 /utf-8 /MP ;-DCMAKE_C_FLAGS_DEBUG=/D_DEBUG /MDd /Z7 /Ob0 /Od /RTC1  -MDd;-DCMAKE_C_FLAGS_MISIZEREL=/O1 /Ob1 /DNDEBUG -MD;-DCMAKE_C_FLAGS_RELEASE=/MD /O2 /Oi /Gy /DNDEBUG /Z7  -MD;-DCMAKE_C_FLAGS_RELWITHDEBINFO=/Z7 /O2 /Ob1 /DNDEBUG -MD;-DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON;-DCMAKE_FIND_PACKAGE_NO_PACKAGE_REGISTRY=ON;-DCMAKE_INSTALL_LIBDIR=lib;-DCMAKE_OSX_SYSROOT=;-DCMAKE_VERBOSE_MAKEFILE=ON;-DCMAKE_C_COMPILER_LAUNCHER=C:\Users\admin\git_projects\ccache.exe;-DCMAKE_CXX_COMPILER_LAUNCHER=C:\Users\admin\git_projects\ccache.exe;-DCMAKE_INSTALL_PREFIX=G:/v/b1/arrow/x64-windows-dbg/mimalloc_ep/src/mimalloc_ep;-DMI_OVERRIDE=OFF;-DMI_LOCAL_DYNAMIC_TLS=ON;-DMI_BUILD_OBJECT=OFF;-DMI_BUILD_SHARED=OFF;-DMI_BUILD_TESTS=OFF;-GNinja;-S;G:/v/b1/arrow/x64-windows-dbg/mimalloc_ep-prefix/src/mimalloc_ep;-B;G:/v/b1/arrow/x64-windows-dbg/mimalloc_ep-prefix/src/mimalloc_ep-build

  Invalid character escape '\U'.


ninja: build stopped: subcommand failed.
```
</details>

--- a/ports/arrow/0007-fix-path.patch
+++ b/ports/arrow/0007-fix-path.patch
@@ -1,0 +1,18 @@
+diff --git a/cpp/cmake_modules/ThirdpartyToolchain.cmake b/cpp/cmake_modules/ThirdpartyToolchain.cmake
+index 978f03198..3a6f36350 100644
+--- a/cpp/cmake_modules/ThirdpartyToolchain.cmake
++++ b/cpp/cmake_modules/ThirdpartyToolchain.cmake
+@@ -975,9 +975,11 @@ set(EP_COMMON_CMAKE_ARGS
+ 
+ # Enable s/ccache if set by parent.
+ if(CMAKE_C_COMPILER_LAUNCHER AND CMAKE_CXX_COMPILER_LAUNCHER)
++  file(TO_CMAKE_PATH "${CMAKE_C_COMPILER_LAUNCHER}" EP_CMAKE_C_COMPILER_LAUNCHER)
++  file(TO_CMAKE_PATH "${CMAKE_CXX_COMPILER_LAUNCHER}" EP_CMAKE_CXX_COMPILER_LAUNCHER)
+   list(APPEND EP_COMMON_CMAKE_ARGS
+-       -DCMAKE_C_COMPILER_LAUNCHER=${CMAKE_C_COMPILER_LAUNCHER}
+-       -DCMAKE_CXX_COMPILER_LAUNCHER=${CMAKE_CXX_COMPILER_LAUNCHER})
++       -DCMAKE_C_COMPILER_LAUNCHER=${EP_CMAKE_C_COMPILER_LAUNCHER}
++       -DCMAKE_CXX_COMPILER_LAUNCHER=${EP_CMAKE_CXX_COMPILER_LAUNCHER})
+ endif()
+ 
+ if(NOT ARROW_VERBOSE_THIRDPARTY_BUILD)

--- a/ports/arrow/portfile.cmake
+++ b/ports/arrow/portfile.cmake
@@ -14,6 +14,7 @@ vcpkg_extract_source_archive(
         0004-android-musl.patch
         0005-android-datetime.patch
         0006-cmake-msvcruntime.patch
+        0007-fix-path.patch # From https://github.com/apache/arrow/issues/39023#issuecomment-1835390089
 )
 
 vcpkg_check_features(OUT_FEATURE_OPTIONS FEATURE_OPTIONS

--- a/ports/arrow/vcpkg.json
+++ b/ports/arrow/vcpkg.json
@@ -1,6 +1,7 @@
 {
   "name": "arrow",
   "version": "19.0.0",
+  "port-version": 1,
   "description": "Cross-language development platform for in-memory analytics",
   "homepage": "https://arrow.apache.org",
   "license": "Apache-2.0",

--- a/versions/a-/arrow.json
+++ b/versions/a-/arrow.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "f4ca71d486f873fbaeae4157f88bcf7b25712440",
+      "version": "19.0.0",
+      "port-version": 1
+    },
+    {
       "git-tree": "36b25d8cfc7098069e26a5185c684dddffa7cd47",
       "version": "19.0.0",
       "port-version": 0

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -246,7 +246,7 @@
     },
     "arrow": {
       "baseline": "19.0.0",
-      "port-version": 0
+      "port-version": 1
     },
     "arsenalgear": {
       "baseline": "2.1.1",

